packages/react-label/src/components/Label/useLabel.tsx

```diff
@@ -20,38 +19,25 @@ const mergeProps = makeMergeProps<LabelState>({ deepMerge: labelShorthandProps }
  * @param ref - reference to root HTMLElement of Label
  * @param defaultProps - (optional) default prop values provided by the implementing type
  */
-export const useLabel = (props: LabelProps, ref: React.Ref<HTMLElement>, defaultProps?: LabelProps): LabelState => {
-  const state = mergeProps(
-    {
-      ref,
-      as: 'label',
-      size: 'medium',
-      required: {
-        as: 'span',
-      },
-    },
-    defaultProps && resolveLabelShorthandProps(defaultProps),
-    resolveLabelShorthandProps(props),
-  );
-
-  return state;
-};
-
-/**
- * Label will first need to check if required is a boolean or shorthandprops,
- * this allows for the required prop to handle both the default asterisk for required
- * or a custom required text.
- */
-const resolveLabelShorthandProps = (props: LabelProps) => {
-  let propsNormalized;
-  if (props.required === true) {
-    propsNormalized = { ...props, required: { children: '*' } };
-  } else if (props.required === false) {
-    propsNormalized = { ...props, required: undefined };
+export const useLabel = (props: LabelProps, ref: React.Ref<HTMLElement>): LabelState => {
+  const { disabled = false, required = false, strong = false } = props;
+  let requiredShorthand;
+  if (required === true) {
+    requiredShorthand = { children: '*' };
+  } else if (required === false) {
+    requiredShorthand = undefined;
   } else {
-    // TypeScript needs a nudge to figure out that props.required won't be a boolean here
-    propsNormalized = props as LabelProps & { required?: Exclude<LabelProps['required'], boolean> };
+    requiredShorthand = required;
   }
 
-  return resolveShorthandProps(propsNormalized, labelShorthandProps);
+  const state: LabelState = {
+    disabled,
+    required: resolveShorthand(requiredShorthand),
```

Comment:
```suggestion
    required: resolveShorthand(required, {
      defaultProps: { children: '*' },
      required: !!required,
    }),
```

Can you please check this suggestion? It looks that it should work properly and replace these `if` conditions

Reply:
That fails test, I'll look into that to see what can be done